packages/devtools_app/lib/src/screens/profiler/cpu_profile_controller.dart

```diff
@@ -384,53 +446,74 @@ class CpuProfilerController
     _processingNotifier.value = true;
 
     try {
-      _dataNotifier.value = await processDataForTag(tag);
+      _dataNotifier.value = await processDataForTag(tag).then(
+        (e) => e.getActive(viewType.value),
+      );
     } catch (e) {
       // In the event of an error, reset the data to the original CPU profile.
       final filteredOriginalData = cpuProfileStore.lookupProfile(
         label: _wrapWithFilterSuffix(userTagNone),
       )!;
-      _dataNotifier.value = filteredOriginalData;
+      _dataNotifier.value = filteredOriginalData.getActive(viewType.value);
       throw Exception('Error loading data with tag "$tag": ${e.toString()}');
     } finally {
       _processingNotifier.value = false;
     }
   }
 
-  Future<CpuProfileData> processDataForTag(String tag) async {
+  Future<CpuProfilePair> processDataForTag(String tag) async {
     final profileLabel = _wrapWithFilterSuffix(tag);
     final filteredDataForTag = cpuProfileStore.lookupProfile(
       label: profileLabel,
     );
     if (filteredDataForTag != null) {
       if (!filteredDataForTag.processed) {
-        await transformer.processData(
-          filteredDataForTag,
+        await filteredDataForTag.processData(
+          transformer: transformer,
           processId: profileLabel,
         );
       }
       return filteredDataForTag;
     }
 
-    var data = cpuProfileStore.lookupProfile(label: tag);
+    var data = cpuProfileStore.lookupProfile(
+      label: tag,
+    );
     if (data == null) {
-      final fullData = cpuProfileStore.lookupProfile(label: userTagNone)!;
-      data = CpuProfileData.fromUserTag(fullData, tag);
-      cpuProfileStore.storeProfile(data, label: tag);
+      final fullData = cpuProfileStore.lookupProfile(
+        label: userTagNone,
+      )!;
+      data = CpuProfilePair.fromUserTag(fullData, tag);
+      cpuProfileStore.storeProfile(
+        data,
+        label: tag,
+      );
     }
 
     data = applyToggleFilters(data);
     if (!data.processed) {
-      await transformer.processData(
-        data,
+      await data.processData(
```

Comment:
nit: rename to process now that it is being called on the data itself. `data.processData` is a bit redundant